python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -370,17 +370,41 @@ def get_prev_partition_window(self, start_dt: datetime) -> Optional[TimeWindow]:
     def get_first_partition_window(
         self, current_time: Optional[datetime] = None
     ) -> Optional[TimeWindow]:
-        current_timestamp = (
+        current_time = cast(
+            datetime,
             pendulum.instance(current_time, tz=self.timezone)
             if current_time
-            else pendulum.now(self.timezone)
-        ).timestamp()
+            else pendulum.now(self.timezone),
+        )
+        current_timestamp = current_time.timestamp()
 
         time_window = next(iter(self._iterate_time_windows(self.start)))
-        if time_window.end.timestamp() <= current_timestamp:
-            return time_window
+
+        if self.end_offset == 0:
+            return time_window if time_window.end.timestamp() <= current_timestamp else None
         else:
-            return None
+            if self.end_offset > 0:
```

Comment:
Nitpick: making this an `elif` would save a level of indentation and make this a little easier to read